### PR TITLE
ENH: Clip T2* values at 100ms to keep consistent histogram axes

### LINKED ISSUE #2780: Clip and re-bin T2* histograms
### What would you like to see added in fMRIPrep?

The T2* histograms are set to a certain number of bins, so a single outlier far to the right of the mode will lose a lot of specificity.

t2star_full_diag

I think we should clip at a reasonably high number (100ms?) and let outliers pile up there to indicate something might have gone wrong while preserving some resolution in the modal range.

In the BOLD space there's already a lower bound at 0. I haven't made an effort to clip the resampled images in T1w/standard spaces, which do get some negative values around the edges due to ringing, as I don't think these are used for calculations.

### Do you have any interest in helping implement the feature?

Yes

### Additional information / screenshots

_No response_

### PR DESCRIPTION
**Pull request by effigies**
Closes #2780.

### CODE REVIEW

**Comment by effigies** [comment]
t2star_hist_clipped

@mgxd Any thoughts?

File: fmriprep/data/reports-spec.yml [hunk 1]
```diff
@@ -106,7 +106,8 @@ sections:
     subtitle: T2* map
   - bids: {datatype: figures, desc: t2starhist, suffix: bold}
     caption: A histogram of estimated T2* values within the anatomically-derived gray-matter mask
-      shown in the previous plot.
+      shown in the previous plot. Note that values are clipped at 100ms, so extreme outliers will
+      all appear in the 100ms bin.
```
Comment (mgxd) [inline fmriprep/data/reports-spec.yml:110]: Slight rewording
```suggestion
      shown in the previous plot. Note that values are clipped at 100ms, so any extreme outliers will
      appear in the 100ms bin.
```

**Review by mgxd — APPROVED** [review]
LGTM

**Comment by effigies** [comment]
Only affects ds210, not waiting for ds005 to finish.

**Pull request merged**